#VRML_SIM R2025a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/robots/gctronic/e-puck/protos/E-puck.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/plants/protos/FlowerPot.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/appearances/protos/Parquetry.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/solids/protos/SolidBox.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/robots/gctronic/e-puck/protos/E-puckDistanceSensor.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.5727673508960804 0.5763584092695471 0.5828795294156105 2.080679240519309
  position -5.0102517715309644e-05 -0.041622761418708056 3.827111889628915
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 2.3 2.3
  floorTileSize 2.3 2.3
  floorAppearance Parquetry {
    type "dark strip"
  }
}
E-puck {
  hidden position_0_0 157.69146688961743
  hidden position_0_1 146.15256960883073
  hidden rotation_1 0 -1 0 5.671351097051818
  hidden rotation_2 0 -0.9999999999999999 0 4.643877763479344
  translation -0.88 -0.88 -6.46134e-05
  rotation 0 0 1 1.57
  controller "my_controller_epuck"
  turretSlot [
    Gyro {
      name "gyro(1)"
    }
    GPS {
    }
  ]
  groundSensorsSlot [
    Pen {
      translation 0 0 0.005
      inkColor 0 1 0.5
      leadSize 0.004
    }
    E-puckDistanceSensor {
      name "ds_front"
    }
    E-puckDistanceSensor {
      translation -0.03 0.035 0
      rotation 0 0 1 1.57
      name "ds_left"
    }
    E-puckDistanceSensor {
      translation -0.03 -0.035 0
      rotation 0 0 1 -1.5731853071795863
      name "ds_right"
    }
  ]
}
FlowerPot {
  translation 0.88 0.88 0
}
SolidBox {
  translation -0.07 0 0.1
  size 0.8 0.2 0.2
}
SolidBox {
  translation -0.67 -0.46 0.1
  name "box(1)"
  size 0.4 0.4 0.2
}
SolidBox {
  translation 0.66 0.27 0.1
  name "box(2)"
  size 0.2 0.4 0.2
}
